src/offset.js

```diff
@@ -143,12 +143,19 @@ jQuery.fn.extend({
 		};
 	},
 
+	// This method will return documentElement in the following cases:
+	// 1) For the element inside the iframe, this method will return
+	//    documentElement of the parent window
+	// 2) For the hidden or detached element
+	// 3) For body or html element, i.e. in case of the html node - it will return itself
+	//
+	// but those exceptions were never presented as a real life use-cases
+	// and might be considered as more preferable results
```

Comment:
How about i will add a notice about it, like "But this is undefined and undocumented behavior, could be changed at any point" or something like that.

It feels weird to know about these edge cases, but not put some info about them somewhere.